qt/scientific_interfaces/Indirect/IndirectSettingsPresenter.cpp

```diff
@@ -0,0 +1,119 @@
+  setUpPresenter();
+}
+
+IndirectSettingsPresenter::IndirectSettingsPresenter(
+    IndirectSettingsModel *model, IIndirectSettingsView *view)
+    : QObject(nullptr), m_model(model), m_view(view) {
+  setUpPresenter();
+}
+
+void IndirectSettingsPresenter::setUpPresenter() {
+  connect(m_view.get(), SIGNAL(updateRestrictInputByName(std::string const &)),
+          this, SLOT(updateRestrictInputByName(std::string const &)));
+
+  connect(m_view.get(), SIGNAL(okClicked()), this,
+          SLOT(applyAndCloseSettings()));
+  connect(m_view.get(), SIGNAL(applyClicked()), this, SLOT(applyChanges()));
+  connect(m_view.get(), SIGNAL(cancelClicked()), this, SLOT(closeDialog()));
+
+  initLayout();
+  loadSettings();
+}
+
+void IndirectSettingsPresenter::showDialog() {
+  m_view->show();
+  m_view->setFocus();
+}
+
+void IndirectSettingsPresenter::initLayout() {
+  m_view->setInterfaceSettingsVisible(m_model->hasInterfaceSettings());
+  m_view->setInterfaceGroupBoxTitle(
+      QString::fromStdString(m_model->getSettingsGroup()));
+
+  m_view->setRestrictInputByNameVisible(
+      m_model->isSettingAvailable("restrict-input-by-name"));
+  m_view->setPlotErrorBarsVisible(
+      m_model->isSettingAvailable("plot-error-bars"));
+}
+
+void IndirectSettingsPresenter::applyAndCloseSettings() {
+  saveSettings();
+  closeDialog();
+}
+
+void IndirectSettingsPresenter::applyChanges() {
+  setApplyingChanges(true);
+  saveSettings();
+  setApplyingChanges(false);
+}
+
+void IndirectSettingsPresenter::closeDialog() { m_view->close(); }
+
+void IndirectSettingsPresenter::loadSettings() {
+  m_view->setSelectedFacility(QString::fromStdString(m_model->getFacility()));
+
+  if (m_model->isSettingAvailable("restrict-input-by-name"))
+    m_view->setRestrictInputByNameChecked(
+        getSetting("restrict-input-by-name").toBool());
+
+  if (m_model->isSettingAvailable("plot-error-bars"))
+    m_view->setPlotErrorBarsChecked(getSetting("plot-error-bars").toBool());
+
+  emit applySettings();
+}
+
+QVariant IndirectSettingsPresenter::getSetting(std::string const &settingName) {
+  return m_view->getSetting(QString::fromStdString(m_model->getSettingsGroup()),
+                            QString::fromStdString(settingName));
+}
+
+void IndirectSettingsPresenter::saveSettings() {
+  auto const group = QString::fromStdString(m_model->getSettingsGroup());
+
+  m_model->setFacility(m_view->getSelectedFacility().toStdString());
+
+  if (m_model->isSettingAvailable("restrict-input-by-name"))
+    m_view->setSetting(group, "restrict-input-by-name",
+                       m_view->isRestrictInputByNameChecked());
+
+  if (m_model->isSettingAvailable("plot-error-bars"))
+    m_view->setSetting(group, "plot-error-bars",
+                       m_view->isPlotErrorBarsChecked());
+
+  emit applySettings();
+}
+
+void IndirectSettingsPresenter::updateRestrictInputByName(
+    std::string const &text) {
+  m_view->setRestrictInputByNameChecked(text == "ISIS");
```

Comment:
Can we do better here? I think it would make sense to restrict the inputs by default regardless on the facility.

Reply:
@gvardany @robertapplin ISIS want to use the suffixes; ILL don’t. What is to default going to be? My 40 yrs of experience says that the default should be for suffixes!! A possibility might be to have it defined in a file - Facilities.xml?

Reply:
Yes, that's what I meant. The default should be to use the suffixes regardless of facility.